game-core/src/main/java/games/strategy/engine/data/properties/CollectionProperty.java

```diff
@@ -27,33 +28,33 @@
    */
   public CollectionProperty(final String name, final String description, final Collection<T> values) {
     super(name, description);
-    m_values = new ArrayList<>(values);
+    this.values = new ArrayList<>(values);
   }
 
   @Override
   public Object getValue() {
-    return m_values;
+    return values;
   }
 
   @Override
   @SuppressWarnings("unchecked")
   public void setValue(final Object value) throws ClassCastException {
-    m_values = (List<T>) value;
+    values = (List<T>) value;
```

Comment:
This screams at me "kill me pls" 😵
We really should make those types generic in the future.